service/patch.go

```diff
@@ -69,13 +68,12 @@ func (uis *UIServer) patchPage(w http.ResponseWriter, r *http.Request) {
 	}
 
 	uis.WriteHTML(w, http.StatusOK, struct {
-		ProjectData projectContext
-		User        *user.DBUser
-		Version     *uiVersion
-		Variants    map[string]model.BuildVariant
-		Tasks       []interface{}
-		CanEdit     bool
-	}{projCtx, currentUser, versionAsUI, variantMappings, tasksList, uis.canEditPatch(currentUser, projCtx.Patch)}, "base",
+		Version  *uiVersion
+		Variants map[string]model.BuildVariant
+		Tasks    []interface{}
+		CanEdit  bool
+		ViewData
+	}{versionAsUI, variantMappings, tasksList, uis.canEditPatch(currentUser, projCtx.Patch), uis.GetCommonViewData(w, r, true, true)}, "base",
```

Comment:
Delete `MustHaveUser`? I'm not going to make further comments like these, since either I'm misunderstanding or this just needs a little cleanup.